### PR TITLE
feat(test): add `Document.example()`

### PR DESCRIPTION
**Pull request by jon-chuang**
# Description

speed up dev loop with a semantically meaningful example

TLDR:

```python
from llama_index import Document, VectorStoreIndex
index = VectorStoreIndex.from_documents([Document.example()])

# Try interacting with the document somehow
```

## Type of Change

- [X] New feature (non-breaking change which adds functionality)

Fixes: partial https://github.com/jerryjliu/llama_index/issues/6738

### CODE REVIEW

File: llama_index/schema.py [hunk 1]
```diff
@@ -380,6 +380,17 @@ def to_langchain_format(self) -> LCDocument:
     def from_langchain_format(cls, doc: LCDocument) -> "Document":
         """Convert struct from LangChain document format."""
         return cls(text=doc.page_content, metadata=doc.metadata)
+    
+    @classmethod
+    def for_test() -> "Document":
```
Comment (Disiok) [inline llama_index/schema.py:385]: Is there a reason to prefer doing this over creating a pytest fixture?

Comment (jon-chuang) [reply llama_index/schema.py]: Hmm, not really, but it's not just for unit tests, it's also just convenience when you wanna test something out.

```python
from llama_index import Document, VectorStoreIndex

index = VectorStoreIndex.from_documents([Document.for_test()])
```

Comment (jon-chuang) [reply llama_index/schema.py]: Actually, maybe it should be a more legit document in this case.

Comment (jon-chuang) [reply llama_index/schema.py]: Something that actually has semantic content, like the paul graham essay

Comment (jon-chuang) [reply llama_index/schema.py]: More for testing semantically-relevant features. E.g. testing the GraphIndex

Comment (jon-chuang) [reply llama_index/schema.py]: Changed the method to `example` to distinguish the use case from test.

**Pull request merged**